app/controllers/application_controller.rb

```diff
@@ -313,6 +315,37 @@ def store_location(path)
     super
   end
 
+  # all create and update actions should have post parameters. If not, in addition to not doing anything,
+  # they might cause errors where these parameters are expected
+  def validate_contains_post_params
+
+    # post values should be in the form {model => {attr => value, attr2 => value}}
+    # an empty body parsed as json will have the form {model => {}}
+    # json body with application/x-www-form-urlencoded content type will have the form {json_string => {}}
+    # json or form encoded with text/plain content type will have the form {}
+    if request.POST.values.all? { |val| val.nil? || (val.is_a?(Hash) && val.empty?) }
+
+      if (request.body.string == "")
```

Comment:
More idiomatic ruby would use `.blank?` or `.empty?`. Also I believe the parentheses are not needed here